### ISSUE TITLE
event system

### ISSUE THREAD

**Issue by 1isenok** [suriyun-production/mmorpg-kit-docs#174]
v 3d mmorpg kit yest' zamechatel'naya vozmozhnost' zagruzki intsans, no net vozmozhnosti upravleniya sobytiyami proiskhodyashchimi v instans, ya imeyu vvidu, dlya primera net upravleniya dvizheniyem monstrov v opredelennom napravlenii, naprimer monstry poyavlyayutsya raz v 2 minuty i begut srazhat'sya drug s drugom ili ukazat' srazheniye za ob'yekt, flag i t.p. ili khotya by vremya zhizni instans, naprimer ya zakhozhu na kartu i yest' 3 minuty poka instans ne zakroyetsya, chto by ya uspel ubit' opredelennoye kollvo mobov, chto by ogranichit' igrokov
the 3d mmorpg kit has a wonderful ability to load intansance, but there is no way to manage events occurring in the instance, I mean, for example, there is no control over the movement of monsters in a certain direction, for example, monsters appear every 2 minutes and run to fight each other or indicate a battle for object, flag, etc. or at least the instance lifetime, for example, I go to the map and have 3 minutes until the instance closes, so that I can kill a certain number of mobs in order to limit the players

**Comment by insthync** [comment]
Ahh, I will add example codes in [ScriptsOnDemand](https://github.com/insthync/UnityMultiplayerARPG_ScriptsOnDemand) repo.
Scripts list:
* Monster spawning with timer, about this component I not sure that all monsters have to died then the timer continues running or not?
* Instance lifetime manager.

**Comment by 1isenok** [comment]
No, I don't mean the mobs life timer. The map instance life timer, for example, I appear in instance and I have a certain time to collect gold or other resources. It is also very useful if this is an instance map where battles for capturing the flag take place (territories, etc., for example, a limited time is given for capturing the flag, who captured the flag / held at the indicated time, received the territory). Or for example a map instance for passing, you need to kill the boss and his henchmen for a limited amount of time. Based on the time parameter, you can also introduce many advantages, such as achievements, made quickly received - 100 gold, made slowly received 10, as an example.


those. instance map, how the event ceases to exist after the timer ends

one more example, a player enters an instance, fights, does his own thing, as soon as the timer stops at 0:00, the instance closes and the player teleports to respawn point

**Comment by 1isenok** [comment]
and monster control about which I wrote above, it means that the monsters would create an imitation of battle, fight with each other, there is the parameter "ally id" with it, as I understand it, you can set different clans to monsters and make them fight among themselves, this can add some interest, for those who will make difficult cards for passing or options for starting players, play for monsters or for people. sorry my english is very bad

**Comment by lophak** [comment]
I found your idea very interesting, timed instances where there are challenges similar to what you see in some strategy games like DOFUS, the character has X minutes to complete the instance where he must defeat X enemies and defeat the BOSS, if he achieves this feat within the time he receives the full prize, if he finishes with a lot of time to spare he receives a bigger bonus or even a title, if he cannot finish before the time is up the instance closes without having to take any action.
